recipes/sqlite3/all/CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.7)
+cmake_minimum_required(VERSION 3.18)
```

Comment:
well, it's too much I thunk, there are other ways to have REQUIRED behavior.

Reply:
Is there a problem with asking for a newer CMake version?
I can manually error out with `message(FATAL_ERROR "bla")` if that's the case, but I think it is cleaner to use the CMake built ins.

Reply:
The CI build error already tells me that asking for 3.18 is too much. I'll do it manually then.
How should we wrap the find_library() call? Talking about your other comment
https://github.com/conan-io/conan-center-index/pull/9235#discussion_r797844444

Reply:
Somethink like this:

```cmake
function(find_library_required foundVar framework)
    find_library(${foundVar} ${framework})
    if(NOT ${foundVar})
        message(FATAL_ERROR "${framework} framework not found")
    endif()
endfunction()
```

But what you are trying to achieve is not portable.

Reply:
I'm not super familiar with this, so I was expecting to be able to do something like
```
if (LINUX)
  find_library(m REQUIRED)
endif()
```
But yeah, I agree that the gain is minimal. I didn't considered the rest of the platforms.
If you agree I'll decline the PR.